src/main/scala/sigmastate/utxo/transformers.scala

```diff
@@ -180,46 +180,34 @@ case class ForAll[IV <: SType](input: Value[SCollection[IV]],
 }
 
 
-case class Fold[IV <: SType](input: Value[SCollection[IV]],
+case class Fold[IV <: SType, OV <: SType](input: Value[SCollection[IV]],
                              id: Byte,
-                             zero: Value[IV],
+                             zero: Value[OV],
                              accId: Byte,
```

Comment:
@ergomorphic I found `id` and  `accId` names and/or order confusing. Please confirm their correctness.

Reply:
@greenhat yes, but let's leave is they are. I'm working towards adding Function node in the nearest future, to have first class lambdas, so all this collection ops will be refactored.